### PR TITLE
refactor(rand): remove randombytes_into usages

### PR DESCRIPTION
**Pull request by kurnevsky**

### CODE REVIEW

File: tox_core/src/state_format/old.rs [hunk 1]
```diff
@@ -615,12 +614,13 @@ impl ToBytes for State {
 mod tests {
     use super::*;
 
+    use rand::thread_rng;
     use tox_packet::ip_port::*;
 
     encode_decode_test!(
         tox_crypto::crypto_init().unwrap(),
```
Comment (kpp) [inline tox_core/src/state_format/old.rs:621]: Do we need crypto_init anymore?

File: tox_packet/src/messenger/conference/invite.rs [hunk 2]
```diff
@@ -73,6 +73,6 @@ mod tests {
     encode_decode_test!(
         tox_crypto::crypto_init().unwrap(),
         invite_encode_decode,
-        Invite::new(1, ConferenceType::Text, ConferenceUid::random())
+        Invite::new(1, ConferenceType::Text, ConferenceUid([42; CONFERENCE_UID_BYTES]))
```
Comment (kpp) [inline tox_packet/src/messenger/conference/invite.rs:76]:
```suggestion
        Invite::new(1, ConferenceType::Text, ConferenceUid([42; ConferenceUid::BYTES]))
```
?

File: tox_packet/src/friend_connection/mod.rs [hunk 3]
```diff
@@ -78,7 +78,7 @@ mod tests {
     encode_decode_test!(
         tox_crypto::crypto_init().unwrap(),
         packet_friend_requests_encode_decode,
-        Packet::FriendRequests(FriendRequests::new(NoSpam::random(), vec![1,2,3,4]))
+        Packet::FriendRequests(FriendRequests::new(NoSpam([42; NOSPAMBYTES]), vec![1,2,3,4]))
```
Comment (kpp) [inline tox_packet/src/friend_connection/mod.rs:81]:
```suggestion
        Packet::FriendRequests(FriendRequests::new(NoSpam([42; NoSpam::BYTES]), vec![1,2,3,4]))
```
?

File: tox_core/src/state_format/old.rs (hunk 1 above)
Comment (kurnevsky) [reply tox_core/src/state_format/old.rs:621]: We have it everywhere in such tests, even when it's unneeded. I'll remove it completely in the next MR where libsodium is completely removed.

**Review by kpp — APPROVED** [review]
LGTM except for $STRUCTBYTES consts

File: tox_packet/src/messenger/conference/invite.rs (hunk 2 above)
Comment (kurnevsky) [reply tox_packet/src/messenger/conference/invite.rs:76]: We can change it to this format. but then we need to review other consts as well. And I wouldn't call it `BYTES`, but `SIZE` or similar.

**Pull request merged**